contracts/core/Allo.sol

```diff
@@ -1,79 +1,283 @@
+        feePercentage = _feePercentage;
+
+        emit RegistryUpdated(_registry);
+        emit TreasuryUpdated(_treasury);
+        emit FeeUpdated(_feePercentage);
+
+        __Multicall_init();
     }
 
-    // @todo insert clonable strategy library, including validation that an existing pool has safe strategy
+    /// ====================================
+    /// =========== Modifier ===============
+    /// ====================================
+
+    modifier isPoolAdmin(uint256 _poolId) {
+        if (!registry.isOwnerOrMemberOfIdentity(pools[_poolId].identityId, msg.sender)) {
+            revert NO_ACCESS_TO_ROLE();
+        }
+        _;
+    }
 
-    // creates pool locally, transfers pool amount to distribution strategy => returns poolId
-    // takes fee from user
-    // validates that the owner is actually allowed to use the identity
-    function createPool(PoolData memory /*_poolData*/ ) external pure returns (uint256) {
-        uint32 _poolId = 0;
+    /// ====================================
+    /// ==== External/Public Functions =====
+    /// ====================================
 
-        // todo: return the poolId? what do we want to return here?
-        return _poolId;
+    /// @notice Returns the pool info
+    /// @param _poolId The id of the pool
+    /// @dev calls out to the registry to get the identity metadata
+    function getPoolInfo(uint256 _poolId) external view returns (Pool memory) {
+        return pools[_poolId];
     }
 
-    // passes _data & msg.sender through to the allocation strategy for that pool
-    // it should return a uint that represents the application
+    /// @notice Creates a new pool
+    /// @param _identityId The identityId of the pool creator in the registry
+    /// @param _allocationStrategy The address of the allocation strategy
+    /// @param _distributionStrategy The address of the distribution strategy
+    /// @param _token The address of the token that the pool is denominated in
+    /// @param _amount The amount of the token to be deposited into the pool
+    /// @param _metadata The metadata of the pool
+    function createPool(
+        bytes32 _identityId,
+        address _allocationStrategy,
+        address payable _distributionStrategy,
+        address _token,
+        uint256 _amount,
+        Metadata memory _metadata
+    ) external payable returns (uint256 poolId) {
+        if (!registry.isOwnerOrMemberOfIdentity(_identityId, msg.sender)) {
+            revert NO_ACCESS_TO_ROLE();
+        }
+
+        Pool memory pool = Pool({
+            identityId: _identityId,
+            allocationStrategy: IAllocationStrategy(Clone.createClone(_allocationStrategy, _nonce++)),
+            distributionStrategy: IDistributionStrategy(Clone.createClone(_distributionStrategy, _nonce++)),
+            metadata: _metadata
+        });
+
+        poolId = _poolIndex++;
+
+        if (_amount > 0) {
+            // TODO: Ask product, should we have minimum fee
```

Comment:
@codenamejason

Reply:
Documenting what we discussed in standup today:

We want to have two fee settings on the contract, which will both be controllable via Gitcoin DAO governance votes:
- **protocol fee** — this should be a percentage that is taken out of the pool amount. E.g. a 2% fee on a 100 DAI pool would yield 2 DAI to the protocol fee and 98 DAI to be distributed via the pool. The owner would need to deposit a total of 100 DAI into the pool.
- **minimum protocol fee** — this should be a flat amount. If the owner creates a pool where the calculated protocol fee is less than this minimum amount, then they will be required to pay this minimum fee.

Both of these values should be set to zero to start.

Reply:
@thelostone-mc The only thing I can think of is requiring the min fee to be sent in ETH in addition to the funding amount. So it would be "base fee" as opposed to "min fee", and you'd probably just keep it at 0.

Reply:
Fixed via 1aaaf57